src/components/invoices/Metadatas.tsx

```diff
@@ -32,18 +33,40 @@ gql`
       value
     }
   }
-`
 
-interface MetadatasProps {
-  customer: CustomerMetadatasForInvoiceOverviewFragment
-  invoice: InvoiceMetadatasForInvoiceOverviewFragment
-}
+  query getInvoiceMetadatas($id: ID!) {
+    invoice(id: $id) {
```

Comment:
Something I tend to do all the time on the first model (at least) requested on a query is to hardcode the `id` property, to help the cache storing those results.
We don't have any rules enforcing such not convention in the team, but if you also feel that's relevant, even if the id is maybe spread in one of the fragments bellow, maybe you can consider adding the `id` here

Reply:
I also need to dig into store behaviour cause not having the id probably works perfectly fine. Just IIRC values are stored by `MODEL_NAME:ID`